File: MPChartExample/src/com/xxmassdeveloper/mpchartexample/custom/MyFillFormatter.java
package com.xxmassdeveloper.mpchartexample.custom;

import com.github.mikephil.charting.formatter.IFillFormatter;
import com.github.mikephil.charting.interfaces.dataprovider.LineDataProvider;
import com.github.mikephil.charting.interfaces.datasets.ILineDataSet;

/**
 * Created by Philipp Jahoda on 12/09/15.
 */
public class MyFillFormatter implements IFillFormatter
{

    private float mFillPos = 0f;

    public MyFillFormatter(float fillpos) {
        this.mFillPos = fillpos;
    }

    @Override
    public float getFillLinePosition(ILineDataSet dataSet, LineDataProvider dataProvider) {
        // your logic could be here
        return mFillPos;
    }
}

File: MPChartExample/src/main/java/com/xxmassdeveloper/mpchartexample/custom/MyFillFormatter.kt
package com.xxmassdeveloper.mpchartexample.custom

import com.github.mikephil.charting.formatter.IFillFormatter
import com.github.mikephil.charting.interfaces.dataprovider.LineDataProvider
import com.github.mikephil.charting.interfaces.datasets.ILineDataSet

@Suppress("unused")
class MyFillFormatter(private val fillPos: Float) : IFillFormatter {
    override fun getFillLinePosition(dataSet: ILineDataSet?, dataProvider: LineDataProvider?): Float {
        // your logic could be here
        return fillPos
    }
}
